idp-binding-service/src/main/java/io/mosip/idp/binding/services/WalletBindingServiceImpl.java

```diff
@@ -0,0 +1,241 @@
+ */
+package io.mosip.idp.binding.services;
+
+
+
+import static io.mosip.idp.core.util.Constants.UTC_DATETIME_PATTERN;
+import static io.mosip.idp.core.util.ErrorConstants.AUTH_FAILED;
+import static io.mosip.idp.core.util.ErrorConstants.INVALID_PUBLIC_KEY;
+import static io.mosip.idp.core.util.ErrorConstants.JSON_PROCESSING_ERROR;
+import static io.mosip.idp.core.util.IdentityProviderUtil.ALGO_SHA_256;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.jose4j.jwe.ContentEncryptionAlgorithmIdentifiers;
+import org.jose4j.jwe.JsonWebEncryption;
+import org.jose4j.jwe.KeyManagementAlgorithmIdentifiers;
+import org.jose4j.jwk.RsaJsonWebKey;
+import org.jose4j.lang.JoseException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.mosip.idp.binding.dto.BindingTransaction;
+import io.mosip.idp.binding.entity.IdTokenMapping;
+import io.mosip.idp.binding.entity.PublicKeyRegistry;
+import io.mosip.idp.binding.repository.IdTokenMappingRepository;
+import io.mosip.idp.binding.repository.PublicKeyRegistryRepository;
+import io.mosip.idp.core.dto.AuthChallenge;
+import io.mosip.idp.core.dto.KycAuthRequest;
+import io.mosip.idp.core.dto.KycAuthResult;
+import io.mosip.idp.core.dto.WalletBindingRequest;
+import io.mosip.idp.core.dto.WalletBindingResponse;
+import io.mosip.idp.core.exception.IdPException;
+import io.mosip.idp.core.exception.InvalidAuthFactorTypeException;
+import io.mosip.idp.core.exception.InvalidIndividualIdException;
+import io.mosip.idp.core.exception.InvalidTransactionException;
+import io.mosip.idp.core.exception.KycAuthException;
+import io.mosip.idp.core.spi.AuthenticationWrapper;
+import io.mosip.idp.core.spi.WalletBindingService;
+import io.mosip.idp.core.util.Constants;
+import io.mosip.idp.core.util.IdentityProviderUtil;
+import io.mosip.kernel.signature.dto.JWTSignatureRequestDto;
+import io.mosip.kernel.signature.dto.JWTSignatureResponseDto;
+import io.mosip.kernel.signature.service.SignatureService;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Service
+public class WalletBindingServiceImpl implements WalletBindingService {
+
+	@Autowired
+	private CacheUtilService cacheUtilService;
+
+	@Autowired
+	private AuthenticationWrapper authenticationWrapper;
+
+	@Autowired
+	private PublicKeyRegistryRepository publicKeyRegistryRepository;
+
+	@Autowired
+	private IdTokenMappingRepository idTokenMappingRepository;
+
+	private SignatureService signatureService;
+
+	private ObjectMapper objectMapper;
```

Comment:
I think we need to autowire objectmapper and signatureService

Reply:
fixed